sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -161,6 +175,43 @@ export class ApplicationController extends BaseController {
     const student = await this.studentService.getStudentByUserId(
       userToken.userId,
     );
+    /** Validation for application overlapping dates or Pending PIR.
+     * This validation can be disabled by setting BYPASS_VALIDATIONS_AT_LOCAL to true in .env file.
+     * */
+
+    if (!this.config.bypassValidationsAtLocal) {
+      const existingOverlapApplication =
+        await this.applicationService.validatePIRAndDateOverlap(
+          userToken.userId,
+          subtractDays(studyStartDate, 1),
```

Comment:
As we talked, the BETWEEN in Postgres is inclusive and maybe there is something else to be checked. So, the subtractDays and addDays would not be necessary here.

Reply:
As discussed, the between function did include the date when we match with exact start date or end date. It was an issue with Postgres and Node running in different time zones. Thank you for helping to sort the issue out.